app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt

```diff
@@ -75,12 +78,69 @@ class AutoCompleteApi @Inject constructor(
             .toObservable()
 
     private fun getAutoCompleteBookmarkResults(query: String) =
-        bookmarksDao.bookmarksByQuery("%$query%")
+        bookmarksDao.bookmarksObservable()
+            .map { rankBookmarks(query, it) }
             .flattenAsObservable { it }
             .map {
-                AutoCompleteBookmarkSuggestion(phrase = it.url, title = it.title ?: "", url = it.url)
+                AutoCompleteBookmarkSuggestion(phrase = it.url.toUri().baseHost.orEmpty(), title = it.title.orEmpty(), url = it.url)
```

Comment:
Using the domain as the phrase causes a bug when you try to edit a bookmark result.

Once you see a bookmark result, if you click on the arrow icon the expected behaviour is for the URL to be added in the search bar so users can tweak it. The logic we have right now uses the phrase. We should change this behaviour so it adds the URL rather than the phrase when the result is a bookmark.

In the `BrowserTabFragment` line 881 when we set the `editableSearchClickListener` we should now differentiate between search results and bookmarks and pass the phrase or URL depending on that.